tests/ui/lint/issue-109152.stderr

```diff
@@ -0,0 +1,23 @@
+error: `Iterator::map` call that discard the iterator's values
+  --> $DIR/issue-109152.rs:5:21
+   |
+LL |     vec![42].iter().map(drop);
+   |                     ^^^^----^
+   |                     |   |
+   |                     |   this function returns `()`, which is likely not what you wanted
```

Comment:
Use `args[0].span` as the default span seems reasonable and code is more clean.
@compiler-errors  how do you think.